hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/OnlineDsAssignmentStrategy.java

```diff
@@ -132,7 +132,9 @@ public JpaAction createTargetAction(final String initiatedBy, final TargetWithAc
             final List<JpaTarget> targets, final JpaDistributionSet set) {
         final JpaAction result = super.createTargetAction(initiatedBy, targetWithActionType, targets, set);
         if (result != null) {
-            if (isUserConsentEnabled() && targetWithActionType.isConfirmationRequired()) {
+            final boolean confirmationRequired = targetWithActionType.isConfirmationRequired()
+                    && result.getTarget().getAutoConfirmationStatus() == null;
```

Comment:
At this point we are checking if the confirmation is required and make the check if auto confirmation is active. Only if auto-conf deactivated and confirmation required case we should put the action in WFC state. Otherwise, we will put it directly in the RUNNING state.